Remove main to master sync workflow (#1361)
PennyLaneAI/pennylane-lightning: .github/CHANGELOG.md

- Fixed a security issue in the triggering mechanism of one of the GH Actions scripts.
  [(#1359)](https://github.com/PennyLaneAI/pennylane-lightning/pull/1359)

> - Change default branch references from `master` to `main`. Removed daily master to main sync.

Reviewer: ```suggestion
- Updated default branch references from `master` to `main` and removed the daily sync between them.
```
